Fixed 'Add tags'. Added 'Edit tags'

Theodorson opened the pull request:
I made a fix for "Add"  which can cause problems in the current implementation, if we want to add tags for multiple issues.

Example: 
- first issue with tag1 tag2
- second issue with tag3, tag4
- we use "Add" tags and add the tag5
- result => first issue will have tag1,tag2,tag3,tag4,tag5 and second issue same tags as first issue
- in conclusion the "Add" behaves like "Set"
- new implementation will just add tag5 for both issues
- new result => first issue with tag1,tag2,tag5 and second issue tag3,tag4,tag5

New implementation contains "Add" which now behaves like an add and a new menu entry "Edit (for single issue)" which has same behavior like old "Add" implementation but just for a single issue (useful for easy add/delete tags for one issue).

New "Edit (for single issue)" (old "Add" but just for single issue).
![image](https://user-images.githubusercontent.com/48293392/181176624-602edcd5-38f6-421a-bf82-91f5e66c4401.png)
![image](https://user-images.githubusercontent.com/48293392/181177457-ffbb2d35-d474-48f2-ba5e-9b013914cc4b.png)

New "Add".
![image](https://user-images.githubusercontent.com/48293392/181176891-dda78c58-14ec-43e6-828a-465018d48aa6.png)
![image](https://user-images.githubusercontent.com/48293392/181176914-e1848949-5325-42b5-bf69-7b3fc7899abe.png)

alexandermeindl commented:
Hi @Theodorson,

you are right, with bulk operations adding tags would be better as changing them.

Maybe a better solution for bulk edit (form) would be, to provide one text field to add tags, and another text field to remove them. I am not sure, if it is required to change tags (replacing existing tags) in bulk edit (form).

To your PR:

- I am not happy with the overwrite parameter, maybe there is a better solution without it?
- there are no tests for the new behavior

Theodorson commented:
Hi @alexandermeindl ,

I can add tests for the new behavior. Can you please let me know what is your proposal for implementing without overwrite parameter ? I am not happy with this too. I tried to do other stuffs but finally I managed this.

Theodorson commented:
Hi @alexandermeindl,

Have you thought of an idea that would be better without the overwrite parameter ? I would like to modify the implementation and create tests for it.

Theodorson commented:
Hi @alexandermeindl,

I've changed the implementation transforming `overwrite` parameter to `append`. This means the new parameter is used just for this case and the existing tests weren't changed. I added tests for new behavior too.

alexandermeindl commented:
Hi @Theodorson,

this looks much better for me. I cannot merge your changes because you do not use main branch for your development. All development on this plugin takes place in main branch. If you provide you changes on main branch, I'll merge it after reviewing it.

Theodorson commented:
Hi @alexandermeindl ,

I've made changes on main branch and create this PR  [#39](https://github.com/AlphaNodes/additional_tags/pull/39).  After reviewing/merge it, the changes will be transfered to `stable` branch ?

alexandermeindl commented:
A new stable release can only be released, if additionals plugins is released, too. Most likely end of this year, a new release will be published.

I close this PR, because you created a new one.
